### PR TITLE
4337: Allow confirmations for not initialized Safes

### PR DESCRIPTION
**Pull request by Uxio0**
- Previously, if a Safe was not deployed no more confirmations were allowed.
- After this PR the owners encoded in the `initCode` will be checked if the Safe is not deployed.

### CODE REVIEW

**Comment by Uxio0** [comment]
@JagoFigueroa for QA contact with @yagopv , as this was reported by him

File: safe_transaction_service/account_abstraction/serializers.py [hunk 1]
```diff
@@ -32,20 +32,26 @@
...
 
     def _get_owners(self, safe_address: ChecksumAddress) -> List[ChecksumAddress]:
         """
         :param safe_address:
-        :return: Valid owners to sign the SafeOperation
+        :return:  `init_code` decoded owners if Safe is not deployed or current blockchain owners if Safe is deployed
         """
-        raise NotImplementedError("Must be implemented in subclass")
+        try:
+            return get_safe_owners(safe_address)
```
Comment (moisses89) [inline safe_transaction_service/account_abstraction/serializers.py:50]: I'm thinking that a better approach would be check if there is `init_code`, as we validate [here](https://github.com/safe-global/safe-transaction-service/blob/229d4bf6a533e1819cb84ccf2c5d26c7a67c5f20/safe_transaction_service/account_abstraction/serializers.py#L128) When `init_code` is missing means not deployed address.
At this way we will reduce the number of calls to the RPC node.

File: safe_transaction_service/account_abstraction/serializers.py [hunk 2]
```diff
@@ -363,6 +352,13 @@ def validate(self, attrs):
                 f"SafeOperation with hash={safe_operation_hash_hex} does not exist"
             )
 
+        # Parse valid owners from init code
+        if user_operation_model.init_code:
+            decoded_init_code = decode_init_code(
+                bytes(user_operation_model.init_code), self.ethereum_client
+            )
+            self._deployment_owners = decoded_init_code.owners
```
Comment (moisses89) [inline safe_transaction_service/account_abstraction/serializers.py:360]: Thinking in the future what do you think about to have something like a `dataclass` for the `init_code` and refactor this, I could see that we are doing the same here [safe_transaction_service/account_abstraction/serializers.py:146](https://github.com/safe-global/safe-transaction-service/blob/229d4bf6a533e1819cb84ccf2c5d26c7a67c5f20/safe_transaction_service/account_abstraction/serializers.py#L146)

File: safe_transaction_service/account_abstraction/serializers.py [hunk 3]
```diff
@@ -32,20 +32,26 @@
 # ================================================ #
 class SafeOperationSignatureValidatorMixin:
     """
-    Mixin class to validate SafeOperation signatures. _get_owners must be overridden to define
+    Mixin class to validate SafeOperation signatures. `_get_owners` can be overridden to define
```
Comment (moisses89) [inline safe_transaction_service/account_abstraction/serializers.py:35]:
```suggestion
    Mixin class to validate SafeOperation signatures. 
```
A developer knows that he can override a function class :)

**Review by moisses89 — APPROVED** [review]
I wrote some suggestions, let me know what do you think.

File: safe_transaction_service/account_abstraction/serializers.py (hunk 1 above)
Comment (Uxio0) [reply safe_transaction_service/account_abstraction/serializers.py:50]: That's not true. You can send `init_code` with the address deployed, that's why I did it this way

File: safe_transaction_service/account_abstraction/serializers.py (hunk 2 above)
Comment (Uxio0) [reply safe_transaction_service/account_abstraction/serializers.py:360]: What do you mean? `Decode` already returns a `dataclass`

File: safe_transaction_service/account_abstraction/serializers.py (hunk 3 above)
Comment (Uxio0) [reply safe_transaction_service/account_abstraction/serializers.py:35]: Haha not always 😉

File: safe_transaction_service/account_abstraction/serializers.py (hunk 2 above)
Comment (moisses89) [reply safe_transaction_service/account_abstraction/serializers.py:360]: You are right, then makes sense to have this field `_deployment_owners` int he `SafeOperationSignatureValidatorMixin`? 
For me doesn't look part of this class, I would say that would be better use directly from the decode init code Dataclass.

Comment (Uxio0) [reply safe_transaction_service/account_abstraction/serializers.py:360]: It has to be there, as the default `_get_owners` function is using it, so it needs to be declared

**Pull request merged**